File: framework/src/play-java/src/main/java/play/routing/RoutingDslComponents.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package play.routing

import play.api.{ ApplicationLoader, BuiltInComponentsFromContext }
import play.api.mvc.{ AnyContent, BodyParser }
import play.core.j.JavaContextComponents

/**
 * RoutingDsl components (for compile-time injection).
 */
trait RoutingDslComponents {
  def defaultBodyParser: BodyParser[AnyContent]
  def javaContextComponents: JavaContextComponents
  def routingDsl: RoutingDsl = new RoutingDsl(defaultBodyParser, javaContextComponents)
}

/**
 * RoutingDsl components from the built in components.
 *
 * @see [[BuiltInComponentsFromContext]]
 * @see [[RoutingDslComponents]]
 */
abstract class RoutingDslComponentsFromContext(context: ApplicationLoader.Context) extends BuiltInComponentsFromContext(context) with RoutingDslComponents

File: framework/src/play-java/src/main/java/play/routing/RoutingDslComponents.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.routing;

import play.components.BodyParserComponents;

/**
 * Java Components for RoutingDsl.
 *
 * <p>Usage:</p>
 * <pre>
 * public class MyComponentsWithRouter extends RoutingDslComponentsFromContext implements HttpFiltersComponents {
 *
 *     public MyComponentsWithRouter(ApplicationLoader.Context context) {
 *         super(context);
 *     }
 *
 *     public Router router() {
 *         // routingDsl method is provided by RoutingDslComponentsFromContext
 *         return routingDsl()
 *              .GET("/path").routeTo(() -&gt; Results.ok("The content"))
 *              .build();
 *     }
 *
 *     // other methods
 * }
 * </pre>
 *
 * @see RoutingDsl
 */
public interface RoutingDslComponents extends BodyParserComponents {

    default RoutingDsl routingDsl() {
        return new RoutingDsl(defaultBodyParser(), javaContextComponents());
    }

}
